src/components/scenes/FioPendingRequestDetailsScene.js

```diff
@@ -66,27 +65,27 @@ export class FioPendingRequestDetailsComponent extends Component<Props, LocalSta
   }
 
   setDefaultWallet (): void {
-    const { onSelectWallet } = this.props
-    const { allowedWallets } = this.getWalletsListData()
-    if (allowedWallets && allowedWallets.length) {
-      onSelectWallet(allowedWallets[0].id, allowedWallets[0].currencyCode)
+    const { onSelectWallet, wallets, selectedFioPendingRequest } = this.props
+    const { chain_code, token_code } = selectedFioPendingRequest.content
+    const confirmTokenCode = token_code && token_code !== chain_code ? token_code : null
+    const walletKeys = Object.keys(wallets)
+    if (confirmTokenCode) {
```

Comment:
The first for loop doesn't check if chain_code === wallet.currencyCode. The if(confirmTokenCode) should be moved into the for loop at line 84. The entire first for loop can be removed completely at line 73.

Essentially, only do one for loop that checks if chain_code matches wallet.currencyCode and then checks if confirmTokenCode matches token_code

Reply:
updated